test/integration/server-selection/server_selection.test.ts

```diff
@@ -0,0 +1,248 @@
+      });
+    });
+
+    context(
+      'operationCount is decremented when the server fails to checkout a connection',
+      function () {
+        it(
+          'is zero after failing to check out a connection for a command',
+          TEST_METADATA,
+          async function () {
+            const server = Array.from(client.topology.s.servers.values())[0];
+            expect(server.s.operationCount).to.equal(0);
+
+            sinon.stub(ConnectionPool.prototype, 'checkOut').callsFake(function (cb) {
+              cb(new Error('unable to checkout connection'), undefined);
+            });
+            const commandSpy = sinon.spy(server, 'command');
+
+            const error = await collection
+              .insertOne({
+                name: 'Joe'
+              })
+              .catch(e => e);
+
+            expect(error).to.exist;
+            expect(error).to.match(/unable to checkout connection/i);
+            expect(commandSpy.called).to.be.true;
+            expect(server.s.operationCount).to.equal(0);
+          }
+        );
+
+        it(
+          'is zero after failing to check out a connection for a getMore',
+          TEST_METADATA,
+          async function () {
+            const cursor = collection.find({}, { batchSize: 1 });
+            await cursor.next();
+
+            const server = Array.from(client.topology.s.servers.values())[0];
+            expect(server.s.operationCount).to.equal(0);
+
+            sinon.stub(ConnectionPool.prototype, 'checkOut').callsFake(function (cb) {
+              cb(new Error('unable to checkout connection'), undefined);
+            });
+            const getMoreSpy = sinon.spy(server, 'getMore');
+
+            const error = await cursor.next().catch(e => e);
+
+            expect(error).to.exist;
+            expect(error).to.match(/unable to checkout connection/i);
+            expect(getMoreSpy.called).to.be.true;
+            expect(server.s.operationCount).to.equal(0);
+
+            sinon.restore();
+            await cursor.close();
+          }
+        );
+
+        it(
+          'is zero after failing to check out a connection for a killCursors',
+          TEST_METADATA,
+          async function () {
+            const cursor = collection.find({}, { batchSize: 1 });
+            await cursor.next();
+
+            const server = Array.from(client.topology.s.servers.values())[0];
+            expect(server.s.operationCount).to.equal(0);
+
+            sinon.stub(ConnectionPool.prototype, 'checkOut').callsFake(function (cb) {
+              cb(new Error('unable to checkout connection'), undefined);
+            });
+            const killCursorsSpy = sinon.spy(server, 'killCursors');
+
+            await cursor.close().catch(err => err);
+            // expect(error).to.exist;
+
+            expect(killCursorsSpy.called).to.be.true;
+            expect(server.s.operationCount).to.equal(0);
+          }
+        );
```

Comment:
I think a good additon would be a test that sets up a retryable error and if I'm not mistaken we might have to spy but the operationCount should go up/down/up/down right?

Reply:
Discussed over slack and we won't add this case.